### ISSUE TITLE
Whonix template fails to update (`Failed to fetch... Invalid response from proxy... Unable to connect Server`)

### ISSUE THREAD

**Issue by bluesteal** [QubesOS/qubes-issues#8388]
R4.1.2

Error on updating whonix-gw-16:

ID: update
Function: pkg.uptodate
Result: False
Comment: Failed to fetch tor-https://deb.debian.org/debian/dists/bullseye/InRelease Invalid response from proxy: HTTP/1.0 500 Unable to connect Server: tinyproxy/1.10.0 Content-Type: text/html Connection: close
[IP: 127.0.0.1 8082]

**Comment by adrelanos** [comment]
Was also reported today in Whonix forums:

* [sys-whonix onionized dom0 update fail on Qubes R4.1 with Qubes-Whonix 17](https://forums.whonix.org/t/sys-whonix-onionized-dom0-update-fail-on-qubes-r4-1-with-qubes-whonix-17/16988)
* [sys-whonix dom0 updates failing on Qubes R4.1 with Qubes-Whonix 16](https://forums.whonix.org/t/sys-whonix-dom0-updates-failing-on-qubes-r4-1-with-qubes-whonix-16/16998)

**Comment by adrelanos** [comment]
I didn't make any changes to the Qubes (dom0) updates mechanism in Qubes-Whonix.

* Did Qubes push some related upgrades to Qubes (dom0) updates / qrexec which have been pushed to Qubes R4.1?
* Is there a test in Qubes openQA which tests Qubes dom0 updates over sys-whonix?

Qubes dom0 updates over sys-whonix work for me in Qubes R4.2 with Qubes-Whonix 17.

**Comment by marmarek** [comment]
I don't think anything specific has changed, but I do observe Whonix updates failing on openQA pretty often, retrying usually helps. This isn't anything new, just usual unreliability of Tor.

**Comment by marmarek** [comment]
See the original issue is about updating whonix itself, not dom0 over sys-whonix.

**Comment by bluesteal** [comment]
So if a non-Tor connection is used it will connect?

**Comment by bluesteal** [comment]
That didn't seem to fix it.

**Comment by DemiMarie** [comment]
> I don't think anything specific has changed, but I do observe Whonix updates failing on openQA pretty often, retrying usually helps. This isn't anything new, just usual unreliability of Tor.

Should updates over Tor be configured to retry failed requests forever?

**Comment by marmarek** [comment]
@adrelanos  I do observe the same unreliability of updates of Whonix 17 on R4.2 too, so upgrading to R4.2+Whonix 17 (as suggested in the forum thread) will not fix it.
Here is a list of failed test runs with this exact error message from last month:
https://openqa.qubes-os.org/tests/77045
https://openqa.qubes-os.org/tests/77224
https://openqa.qubes-os.org/tests/77331
https://openqa.qubes-os.org/tests/77545
https://openqa.qubes-os.org/tests/77757
https://openqa.qubes-os.org/tests/77882
https://openqa.qubes-os.org/tests/78320
https://openqa.qubes-os.org/tests/78371
https://openqa.qubes-os.org/tests/78406
https://openqa.qubes-os.org/tests/78694
https://openqa.qubes-os.org/tests/78871
https://openqa.qubes-os.org/tests/78896
https://openqa.qubes-os.org/tests/78932
(this is mix of R4.1, R4.2, both Whonix 16 and 17, with last 3 beeing R4.2+Whonix 17)

**Comment by marmarek** [comment]
And BTW (related to the forum thread), whatever the fix will turn out to be, I'd expect it can be backported to Whonix 16, which is still supported, right?

**Comment by adrelanos** [comment]
> See the original issue is about updating whonix itself, not dom0 over sys-whonix.

Ok. Let's use this issue for general update issues.

And I'll create a separate ticket for dom0 updates over sys-whonix.

> And BTW (related to the forum thread), whatever the fix will turn out to be,

The problem is, I cannot reproduce this issue. One Qubes R4.2. with Qubes-Whonix 17, dom0 updates over Tor are working fine.

> I'd expect it can be backported to Whonix 16, which is still supported, right?

Supported in a sense:

* if there's security issues discovered in Whonix's code and these are fixable, I'll fix and backport.
* connectivity issues such as this, once diagnosed and fixable can be backported too.
* but I won't be able to answer all the user support inquiries.

> I'd expect it can be backported to Whonix 16, 

In theory, that should be doable. Packages built on bookworm are compatible with bullseye. However, issue here I cannot reproduce the issue. 

No changes on Qubes side as you confirmed. No changes on Whonix side as I can confirm. Yet, dom0 updates over sys-whonix brokenness for some users. So I am at loss what's causing this.

**Comment by DemiMarie** [comment]
@bluesteal: if you retry the update, does the problem go away?  If not, does restarting `sys-whonix` help?

**Comment by adrelanos** [comment]
> > I don't think anything specific has changed, but I do observe Whonix updates failing on openQA pretty often, retrying usually helps. This isn't anything new, just usual unreliability of Tor.
> 
> Should updates over Tor be configured to retry failed requests forever?

Maybe not forever. But 3 (or 5) retries with a bit of retry delay (2 or 5 seconds) would be good. Maybe even run a script inside sys-whonix (which I could write which then would then run "SIGNAL newnym" to get a new Tor circuit).

**Comment by adrelanos** [comment]
Can you upgrade Whonix Templates?

Can you upgrade other Templates over sys-whonix?

**Comment by adrelanos** [comment]
For debugging purposes, please run:

    systemcheck --ip-test

In:

* anon-whonix
* sys-whonix
* Whonix-Gateway Template
* Whonix-Workstation Template

And post the output here in case there are any warnings or errors. Otherwise please confirm if no warnings or errors are shown.

**Comment by adrelanos** [comment]
@bluesteal
> So if a non-Tor connection is used it will connect?

@bluesteal
> That didn't seem to fix it.

What did you try?

**Comment by bluesteal** [comment]
Ok there is an error in whonix-gw and whonix-ws:

systemcheck --ip-test

[systemcheck] UpdatesProxy IP Leak Test: Qubes’ UpdatesProxy was not reachable.
(curl exit code: [22] - [HTTP page not retrieved. The requested url was not
found or returned another error with the HTTP error code being 404
or above.

**Comment by adrelanos** [comment]
This issue could be quite similar to my theory in https://github.com/QubesOS/qubes-issues/issues/8389#issuecomment-1666532553. To prove it, I performed the following steps.

In `whonix-gateway-17` Template:

* First, I confirmed that I can run a normal update without issue. Ok. Then for debugging purposes...
* I deleted all files in `/etc/apt/sources.list.d` folder.
* (Whonix doesn't use `/etc/apt/souces.list` by default.)
* Created a new file `/etc/apt/sources.list.d/blocked.list` with the following content.

```
deb tor+https://deb.blocked.com bookworm main
```

* In summary, my whole Debian APT sources only contained `deb tor+https://deb.blocked.com bookworm main
`. (The `tor+` is pretty much irrelevant for this test.)
* `sudo apt update`
* result:

>  Invalid response from proxy: HTTP/1.1 500 Unable to connect  Server: tinyproxy/1.11.1  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Reading package lists... Done                        
> E: Failed to fetch tor+https://deb.blocked.com/dists/bookworm/InRelease  Invalid response from proxy: HTTP/1.1 500 Unable to connect  Server: tinyproxy/1.11.1  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
> E: Some index files failed to download. They have been ignored, or old ones used instead.

In conclusion, the error message which this ticked as opened with `HTTP/1.0 500 Unable to connect Server: tinyproxy/1.10.0 Content-Type: text/html Connection: close` seems to be correct.

At this point I am not convinced that Qubes or Whonix source code has any bugs being responsible for this issue.

HTTP error code `500` is quite common for server error or notifications of an IP being blocked due to some filter such as blocking Tor exit relays.

In simple terms, the server seamed to have an issue and/or was blocking the IP of the Tor exit relays used by the user.

Therefore there's a good chance, that this issue will be transient. Meaning, it will resolve itself after a while.

I have some ideas to further nail down the issue. One could try to disable certain repositories.

* Debian
* Qubes
* Whonix

And then upgrade only from 1 repository at a time using `sudo apt-get update` inside the VM. This would help to nail down which repository is causing issues. I can certainly provide user documentation on how to do that if needed.

Debian repository should be Tor friendly. If not, one could try to use the onion version but then onions itself also have issues generally due to DOS on the Tor network.

Whonix repository isn't blocking Tor users.

Where is Qubes repository hosted nowadays? Any chance these servers are blocking Tor users? This seems the most likely for me for now since no issue was reported for Non-Qubes-Whonix, which has more users than Qubes-Whonix.

Please let me know if this issue is still reproducible.

**Comment by adrelanos** [comment]
> I have some ideas to further nail down the issue. One could try to disable certain repositories.
> 
> * Debian
> * Qubes
> * Whonix
> 
> And then upgrade only from 1 repository at a time using `sudo apt-get update` inside the VM. This would help to nail down which repository is causing issues. I can certainly provide user documentation on how to do that if needed.

Actually much simpler. When running `sudo apt-get update` inside the VM and then posting the log here, APT's output will easily show which repository is causing issues. Therefore please provide the output of `sudo apt-get update` inside the VM should this issue still persist.

**Comment by DemiMarie** [comment]
> HTTP error code `500` is quite common for server error or notifications of an IP being blocked due to some filter such as blocking Tor exit relays.

This is not a server error.  It is an error from tinyproxy stating that it is unable to connect to the destination domain name.  The fact that retrying generally helps means that this is most likely to be a problem with Tor or Whonix.

I noticed that tinyproxy is _not_ connected to use Tor as a SOCKS5 proxy, and instead relies on the transparent proxying.  This seems more fragile than using the SOCKS proxy.  Would that help?

**Comment by bluesteal** [comment]
sudo apt-get update (in whonix-gw):

E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye/InRelease  Could not connect to 127.0.0.1:8082 (127.0.0.1). - connect (113: No route to host)
E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye-updates/InRelease  Unable to connect to 127.0.0.1:8082:
E: Failed to fetch tor+https://deb.debian.org/debian-security/dists/bullseye-security/InRelease  Unable to connect to 127.0.0.1:8082:
E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye-backports/InRelease  Unable to connect to 127.0.0.1:8082:
E: Failed to fetch tor+https://fasttrack.debian.net/debian/dists/bullseye-fasttrack/InRelease  Could not connect to 127.0.0.1:8082 (127.0.0.1). - connect (113: No route to host)
E: Failed to fetch tor+https://deb.whonix.org/dists/bullseye/InRelease  Could not connect to 127.0.0.1:8082 (127.0.0.1). - connect (113: No route to host)
E: Failed to fetch https://deb.qubes-os.org/r4.1/vm/dists/bullseye/InRelease  Could not connect to 127.0.0.1:8082 (127.0.0.1). - connect (113: No route to host)

**Comment by marmarek** [comment]
Unable to connect to updates proxy this way is interesting, it looks like the either `qubes-updates-proxy-forwarder.socket` is not running (you can check with `systemctl status qubes-updates-proxy-forwarder.socket`), or firewall prevents the connection.
Or maybe `qubes-whonix-torified-updates-proxy-check.service` failed?
@adrelanos is there anything else in Whonix template itself that could prevent reaching localhost tcp port?

**Comment by bluesteal** [comment]
I can check this in dom0? Where to check this?

**Comment by marmarek** [comment]
In whonix-gw where apt-get failed.

**Comment by marmarek** [comment]
You can also call `systemctl` without arguments and see if there is any failed service.

**Comment by bluesteal** [comment]
systemctl status qubes-updates-proxy-forwarder.socket
● qubes-updates-proxy-forwarder.socket - Forward connection to updates proxy over Qubes RPC
     Loaded: loaded (/lib/systemd/system/qubes-updates-proxy-forwarder.socket; enabled; vendor preset: enabled)
     Active: active (listening) since Sat 2023-08-05 21:05:43 UTC; 5h 54min ago
     Listen: 127.0.0.1:8082 (Stream)
   Accepted: 29; Connected: 0;
      Tasks: 0 (limit: 256)
     Memory: 4.0K
        CPU: 17ms
     CGroup: /system.slice/qubes-updates-proxy-forwarder.socket


systemctl:

qubes-whonix-torified-updates-proxy-check.service   loaded active exited    Qubes Whonix Torified Updates Proxy Check

**Comment by bluesteal** [comment]
I should mention that I have two sys-whonix VMs running and one is a cloned VM.

**Comment by adrelanos** [comment]
Could you please use this command to check for failed systemd units. This command is a bit simpler and only shows what is failed.

    sudo systemctl list-units --failed

**Comment by adrelanos** [comment]
In Template, please run:

    sudo journalctl --boot --output cat -u qubes-whonix-torified-updates-proxy-check.service

And.

    ls -la /run/updatesproxycheck

Then please post the output here.

**Comment by bluesteal** [comment]
SYS-WHONIX

sudo systemctl list-units --failed

 UNIT LOAD ACTIVE SUB DESCRIPTION
0 loaded units listed.

sudo journalctl --boot --output cat -u qubes-whonix-torified-updates-proxy-check.service

Starting Qubes Whonix Torified Updates Proxy Check...
Finished Qubes Whonix Torified Updates Proxy Check.

ls -la /run/updatesproxycheck

total 0
drwxr-xr-x  2 updatesproxycheck updatesproxycheck  40 Aug  6 16:45 .
drwxr-xr-x 34 root              root              820 Aug  6 16:45 ..

WHONIX-GW-16

sudo systemctl list-units --failed
UNIT LOAD ACTIVE SUB DESCRIPTION
0 loaded units listed.

sudo journalctl --boot --output cat -u qubes-whonix-torified-updates-proxy-check.service
Starting Qubes Whonix Torified Updates Proxy Check...
Finished Qubes Whonix Torified Updates Proxy Check.

ls -la /run/updatesproxycheck
total 0
drwxr-xr-x  2 updatesproxycheck updatesproxycheck  80 Aug  6 17:11 .
drwxr-xr-x 33 root              root              780 Aug  6 17:11 ..
-rw-r--r--  1 updatesproxycheck updatesproxycheck   0 Aug  6 17:11 whonix-secure-proxy
-rw-r--r--  1 updatesproxycheck updatesproxycheck   0 Aug  6 17:11 whonix-secure-proxy-check-done

**Comment by adrelanos** [comment]
@DemiMarie https://github.com/QubesOS/qubes-issues/issues/8388#issuecomment-1666591662

> I noticed that tinyproxy is _not_ connected to use Tor as a SOCKS5 proxy, and instead relies on the transparent proxying. This seems more fragile than using the SOCKS proxy. Would that help?

That is generally a good idea. Created

* https://github.com/QubesOS/qubes-issues/issues/8398

for it. However, I doubt that is the issue here.

Only 1 user for now has been reporting persistent Qubes-Whonix template upgrade issues. And the bug report is incomplete. It doesn't contain all the steps how to reproduce the issue. Therefore I am suspecting a corner case, perhaps coming from removing too many packages or other complex stuff such as modifying firewall rules.

----

Many improvements to systemcheck [`check_qubes.bsh`](https://github.com/Kicksecure/systemcheck/blob/master/usr/libexec/systemcheck/check_qubes.bsh) have been made.

* Checking the files in `/run/updatesproxycheck` (looking at `qubes-whonix-torified-updates-proxy-check.service`) will be automated.
* Local (no external network connection) Qubes UpdatesProxy reachablity will be routinely tested when running systemcheck.
* Remote (external network connection) Qubes UpdatesProxy connectivity will be checked when using `--leak-tests`.

These systemcheck improvements flow into Whonix 17 as per usual package migration. Usage as per usual:

    systemcheck --verbose --leak-tests

----

If other users can persistently reproduce the same issue, please post here with instructions on how to reproduce the issue.

**Comment by adrelanos** [comment]
This very comment is technical and only for developers.

@bluesteal wrote:
> WHONIX-GW-16
> 
> sudo systemctl list-units --failed
> UNIT LOAD ACTIVE SUB DESCRIPTION
> 0 loaded units listed.
> 
> sudo journalctl --boot --output cat -u qubes-whonix-torified-updates-proxy-check.service
> Starting Qubes Whonix Torified Updates Proxy Check...
> Finished Qubes Whonix Torified Updates Proxy Check.
> 
> ls -la /run/updatesproxycheck
> total 0
> drwxr-xr-x 2 updatesproxycheck updatesproxycheck 80 Aug 6 17:11 .
> drwxr-xr-x 33 root root 780 Aug 6 17:11 ..
> -rw-r--r-- 1 updatesproxycheck updatesproxycheck 0 Aug 6 17:11 whonix-secure-proxy
> -rw-r--r-- 1 updatesproxycheck updatesproxycheck 0 Aug 6 17:11 whonix-secure-proxy-check-done

Both status files `/run/updatesproxycheck/whonix-secure-proxy-check-done` and `/run/updatesproxycheck/whonix-secure-proxy` exist.

This means that `qubes-whonix-torified-updates-proxy-check.service` / `/usr/lib/qubes-whonix/init/torified-updates-proxy-check` has run `UWT_DEV_PASSTHROUGH=1 curl --silent http://127.0.0.1:8082/` and the output of that command included `<meta name="application-name" content="tor proxy"/>`. In simpler terms, Qubes UpdatesProxy on `sys-whonix` was reachable.

Qubes UpdatesProxy on a Whonix-Gateway App Qube with certainty, because otherwise it would not be `tor proxy`.

@marmarek 
> Unable to connect to updates proxy this way is interesting, it looks like the either `qubes-updates-proxy-forwarder.socket` is not running (you can check with `systemctl status qubes-updates-proxy-forwarder.socket`), or firewall prevents the connection. Or maybe `qubes-whonix-torified-updates-proxy-check.service` failed?

It seems like we currently don't have this issue as per above.

> @adrelanos is there anything else in Whonix template itself that could prevent reaching localhost tcp port?

Just to address this point even though only theoretic. Unfortunately, yes. References:

* https://www.whonix.org/wiki/Stream_Isolation/Easy
* uwt, torsocks
* workarounds:
  * `export UWT_DEV_PASSTHROUGH="1"`
  * or
  * `UWT_DEV_PASSTHROUGH curl ...`

But not the issue here currently.

**Comment by adrelanos** [comment]
This very comment is technical and only for developers.

Furthermore, it's unlikely that the torified UpdatesProxy check failed. Even without systemcheck checking for it, precautions have been made years ago. For demonstration, debugging purposes:

1) Simulate torified UpdatsProxy check failed.

`sudo rm /run/updatesproxycheck/whonix-secure-proxy`

2) Run apt.

`sudo apt update`

Result:

```
WARNING: Execution of /usr/bin/apt prevented by /etc/uwt.d/40_qubes.conf because no torified Qubes updates proxy found.
Please make sure Whonix-Gateway (commonly called sys-whonix) is running.

Check your _dom0_ settings in the /etc/qubes/policy.d/ folder.

To see if it is fixed, try running in Whonix Template:

sudo systemctl restart qubes-whonix-torified-updates-proxy-check

Then try to update / use apt-get again.

For more help on this subject see:
https://www.whonix.org/wiki/Qubes/UpdatesProxy

If this warning message is transient, it can be safely ignored.
```

Thanks to [`/etc/uwt.d/40_qubes.conf`](https://github.com/Whonix/qubes-whonix/blob/master/etc/uwt.d/40_qubes.conf) which checks if `/run/updatesproxycheck/whonix-secure-proxy` exists which only exist does if Qubes UpdatesProxy was reachable and the output of the test included `tor proxy`.

**Comment by adrelanos** [comment]
This very comment is technical and only for developers.

@adrelanos 
> In conclusion, the error message which this ticked as opened with `HTTP/1.0 500 Unable to connect Server: tinyproxy/1.10.0 Content-Type: text/html Connection: close` seems to be correct.

@DemiMarie 
> > HTTP error code `500` is quite common for server error or notifications of an IP being blocked due to some filter such as blocking Tor exit relays.
> 
> This is not a server error. It is an error from tinyproxy stating that it is unable to connect to the destination domain name.

To test this theory,

* 1) I've run in `sys-whonix`: `sudo systemctl stop tor@default` and confirmed in journal log that it's really stopped.
* 2) Run `sudo apt update` in `whonix-gateay-17` Template.

The result included:

>  Invalid response from proxy: HTTP/1.1 500 Unable to connect  Server: tinyproxy/1.11.1  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]

So you are right, it could be that tinyproxy stating that it is unable to connect to the destination domain name.

However, for debugging purposes, let's add `/etc/apt/sources.list.d/blocked.list` with contents:

    deb tor+https://deb.blocked.com bookworm main

Result:

All APT sources are functional except `blocked.com` (to be expected).

> Ign:1 tor+https://deb.blocked.com bookworm InRelease                                                                                                         
>   Invalid response from proxy: HTTP/1.1 500 Unable to connect  Server: tinyproxy/1.11.1  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
> Reading package lists... Done
> E: Failed to fetch tor+https://deb.blocked.com/dists/bookworm/InRelease  Invalid response from proxy: HTTP/1.1 500 Unable to connect  Server: tinyproxy/1.11.1  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]

This means, at the level of output by `apt` in Template, from what tinyproxy is telling us:

We cannot distinguish from between:

* A) tinyproxy stating that it is unable to connect to the destination domain name, and
* B) connection blocked by a remote APT server.

The symptom / output is really the same.

**Comment by adrelanos** [comment]
@bluesteal (or other users affected by this issue):

Please run in:

1) Template
2) sys-whonix

`systemcheck --verbose --leak-tests`

And then please post the output here.

**Comment by bluesteal** [comment]
Whonix-GW Warnings:

WARNING: [systemcheck] Hardened Malloc: Disabled
WARNING: [systemcheck] UpdatesProxy IP Leak Test: Qubes' UpdatesProxy was not reachable.
WARNING [systemcheck] Debian Package Update Check Result: Could not check for software updates! (apt-get code: 100)
Please manually check inside this TemplateVM ('whonix-gw-16').
1. Open a terminal. (dom0 -> Start Menu -> Template: whonix-gw-16 -> Terminal)
2. Update. upgrade-nonroot

**Comment by bluesteal** [comment]
Whonix-GW Entire OUTPUT:

[INFO] [systemcheck] whonix-gw-16 | Whonix-Gateway | TemplateVM | Sun 06 Aug 2023 06:51:51 PM UTC
[INFO] [systemcheck] Check sudo Result: OK
[INFO] [systemcheck] Whonix build version: 3:8.1-1
[INFO] [systemcheck] whonix-gateway-packages-dependencies-cli: 23.0-1
[INFO] [systemcheck] derivative_major_release_version /etc/whonix_version: 16
[INFO] [systemcheck] Whonix Support Status of this Major Version: Ok.
[WARNING] [systemcheck] Hardened Malloc: Disabled.
[INFO] [systemcheck] Spectre Meltdown Test: skipping since spectre_meltdown_check=false, ok.
[INFO] [systemcheck] Package Manager Consistency Check Result: Output of command dpkg --audit was empty, ok.
[INFO] [systemcheck] systemd journal check Result:
AppArmor:
########################################

########################################

secomp:
########################################

########################################

warnings:
########################################
Aug 06 17:11:47 host qubes.StartApp+qubes-run-terminal-dom0[1057]: (xfce4-terminal:1064): dbind-WARNING **: 17:11:47.794: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Aug 06 17:16:52 host qubes.StartApp+qubes-run-terminal-dom0[1057]: (xfce4-terminal:1064): Gtk-WARNING **: 17:16:52.140: Could not load a pixbuf from /org/gtk/libgtk/theme/Adwaita/assets/check-symbolic.svg.
########################################

failed:
########################################
Aug 06 17:09:54 host kernel: tsc: Fast TSC calibration failed
Aug 06 17:09:54 host kernel: Failed to register legacy timer interrupt
Aug 06 17:09:56 host swapon[402]: swapon: /dev/xvdc1: swapon failed: Device or resource busy
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:41 host dbus-daemon[917]: [session uid=1000 pid=914] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Aug 06 17:11:47 host qubes.StartApp+qubes-run-terminal-dom0[1057]: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
Aug 06 17:12:15 host sudo[1193]:     user : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/usr/bin/systemctl list-units --failed
Aug 06 18:51:52 host sudo[2757]: systemcheck : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/bin/systemctl --no-pager --no-block --no-legend --failed
########################################

errors:
########################################
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 00, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 01, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 03, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 04, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 05, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 06, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: ACPI Error: No handler or method for GPE 07, disabling event (20220331/evgpe-839)
Aug 06 17:09:54 host kernel: RAS: Correctable Errors collector initialized.
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:35 host xl[562]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 17:11:36 host sdwdate-pre[555]: + gcc /usr/libexec/sdwdate/sclockadj.c -o /usr/libexec/sdwdate/sclockadj -ldl -D_GNU_SOURCE -Wdate-time -D_FORTIFY_SOURCE=3 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now
Aug 06 17:11:47 host qubes.StartApp+qubes-run-terminal-dom0[1057]: (xfce4-terminal:1064): dbind-WARNING **: 17:11:47.794: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
########################################

ordering cycle:
########################################

########################################

To see this for yourself...
1. Open a terminal. (dom0 -> Start Menu -> Template: whonix-gw-16 -> Terminal)
3. Run. sudo apparmor-info
3. Run. sudo journalctl --boot | grep -i syscall=
4. Run. sudo journalctl --boot | grep -i warn
5. Run. sudo journalctl --boot | grep -i fail
6. Run. sudo journalctl --boot | grep -i error
7. Run. sudo journalctl --boot | grep -i "ordering cycle"

If you know what you are doing, feel free to disable this check.
Create a file /etc/systemcheck.d/50_user.conf and add:
systemcheck_skip_functions+=" check_journal "
[INFO] [systemcheck] Qubes Settings Test Result: Ok, qubes_vm_type is TemplateVM.
[INFO] [systemcheck] Check Kernel Messages Test Result: Found nothing remarkable, ok.
[INFO] [systemcheck] Whonix firewall systemd unit check Result: Ok.
[INFO] [systemcheck] Check setup-dist Result: done, ok.
[INFO] [systemcheck] Check Package Manager Running Result: None running, ok.
[INFO] [systemcheck] Tor Check Result: Skipped, because running in TemplateVM, ok.
[INFO] [systemcheck] Tor Config Check Result: Tor config ok.
[INFO] [systemcheck] Tor Running Check Result: Not running in Qubes-Whonix-Gateway TemplateVM, ok.
[INFO] [systemcheck] Whonix Meta Packages Test Result: Meta package qubes-whonix-gateway installed, ok.
[INFO] [systemcheck] Whonix Unwanted Packages Test Result: None found.
[INFO] [systemcheck] Check Initializer Result: /var/lib/initializer-dist/status-files/first_run_initializer.fail does not exist, ok.
[INFO] [systemcheck] Check Virtualizer Result: Supported Virtualizer Qubes detected, continuing.
systemd-detect-virt result: qubes
[INFO] [systemcheck] PVClock Result: /sys/devices/system/clocksource/clocksource0/current_clocksource exist, is xen. This is non-ideal. A known issue. Contributions happily considered. Read more: https://phabricator.whonix.org/T389
[INFO] [systemcheck] Check Timezone Result: /etc/timezone, Etc/UTC matches Etc/UTC, ok.
[INFO] [systemcheck] Check Timezone Result: /usr/share/zoneinfo/Etc/UTC matches /etc/localtime, ok.
[INFO] [systemcheck] IP Forwarding Result: /proc/sys/net/ipv4/ip_forward is 0, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /var/lib/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.cache/tb/torbrowser_updater_error.log does not exist, ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --fqdn" output is "host.localdomain", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname" output is "host", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --ip-address" output is "127.0.0.1", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --ip-address" output is "localdomain", ok.
[INFO] [systemcheck] Entropy Available Check Result: ok. /proc/sys/kernel/random/entropy_avail: 256
[INFO] [systemcheck] Check nonfree Result: Ok, no nonfree packages found. For more information, see:
https://www.whonix.org/wiki/Avoid_nonfree_software
[INFO] [systemcheck] Whonix APT Repository: Enabled.
When the Whonix team releases BULLSEYE updates,
they will be AUTOMATICALLY installed (when you run apt-get dist-upgrade)
along with updated packages from the Debian team. Please
read https://www.whonix.org/wiki/Trust to understand the risk.
If you want to change this, use:
    sudo whonix_repository
[INFO] [systemcheck] Qubes Update Proxy Reachability Test: Trying to reach Qubes update proxy... PROXY_SERVER: http://127.0.0.1:8082/
[INFO] [systemcheck] Qubes Update Proxy Reachability Result: Ok, torified update proxy reachable.
[INFO] [systemcheck] Qubes Update Proxy Functionality Test: Trying to use Qubes update proxy... PROXY_SERVER: http://127.0.0.1:8082/
[INFO] [systemcheck] Qubes Update Proxy Functionality Result: Failed. (Filtered.) (curl exit code: curl_exit_code | curl status message: [22] - [HTTP page not retrieved. The requested url was not found or returned another error with the HTTP error code being 400 or above. This return code only appears if -f, --fail is used.])
[INFO] [systemcheck] SocksPort IP Leak Test: Testing Tor's SocksPort (SOCKS_PORT_SYSTEMCHECK: 9110)...
[INFO] [systemcheck] SocksPort IP Leak Test: Connected to Tor. 
[INFO] [systemcheck] system default networking IP Leak Test: system default networking IP Leak Test...
[INFO] [systemcheck] system default networking IP Leak Test Test Result: Ok, Tor's TransPort was not reachable.
This is expected in default configuration of TemplateVMs in Qubes R4.0 and above, see https://www.whonix.org/wiki/Dev/Qubes#Network_in_TemplateVMs.
(curl exit code: [6] - [Couldn't resolve host. The given remote host was not resolved.])
[INFO] [systemcheck] UpdatesProxy IP Leak Test: Testing Qubes' UpdatesProxy...
[WARNING] [systemcheck] UpdatesProxy IP Leak Test: Qubes' UpdatesProxy was not reachable.
(curl exit code: [22] - [HTTP page not retrieved. The requested url was not found or returned another error with the HTTP error code being 400 or above. This return code only appears if -f, --fail is used.])
[INFO] [systemcheck] Stream Isolation Test: Ok, skipped, because TransPort test failed! Can not test stream isolation. This is expected on Whonix-Gateway's default configuration.
[INFO] [systemcheck] Debian Package Update Check: Checking for software updates via apt-get... ( Documentation: https://www.whonix.org/wiki/Update )
Ign:1 tor+https://fasttrack.debian.net/debian bullseye-fasttrack InRelease                                           
Ign:2 tor+https://deb.debian.org/debian bullseye InRelease                                                           
Ign:3 tor+https://deb.whonix.org bullseye InRelease                                                                  
Ign:4 https://deb.qubes-os.org/r4.1/vm bullseye InRelease                                                            
Ign:1 tor+https://fasttrack.debian.net/debian bullseye-fasttrack InRelease                                           
Ign:5 tor+https://deb.debian.org/debian bullseye-updates InRelease                                                   
Ign:3 tor+https://deb.whonix.org bullseye InRelease                                                                  
Ign:4 https://deb.qubes-os.org/r4.1/vm bullseye InRelease                                                            
Ign:1 tor+https://fasttrack.debian.net/debian bullseye-fasttrack InRelease                                           
Ign:6 tor+https://deb.debian.org/debian-security bullseye-security InRelease                                         
Ign:3 tor+https://deb.whonix.org bullseye InRelease                                                                  
Ign:4 https://deb.qubes-os.org/r4.1/vm bullseye InRelease                                                            
Err:1 tor+https://fasttrack.debian.net/debian bullseye-fasttrack InRelease                                           
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Ign:7 tor+https://deb.debian.org/debian bullseye-backports InRelease                                                 
Err:3 tor+https://deb.whonix.org bullseye InRelease                                                                  
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Err:4 https://deb.qubes-os.org/r4.1/vm bullseye InRelease                                                            
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Ign:2 tor+https://deb.debian.org/debian bullseye InRelease                                              
Ign:5 tor+https://deb.debian.org/debian bullseye-updates InRelease
Ign:6 tor+https://deb.debian.org/debian-security bullseye-security InRelease
Ign:7 tor+https://deb.debian.org/debian bullseye-backports InRelease
Ign:2 tor+https://deb.debian.org/debian bullseye InRelease
Ign:5 tor+https://deb.debian.org/debian bullseye-updates InRelease
Ign:6 tor+https://deb.debian.org/debian-security bullseye-security InRelease
Ign:7 tor+https://deb.debian.org/debian bullseye-backports InRelease
Err:2 tor+https://deb.debian.org/debian bullseye InRelease
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Err:5 tor+https://deb.debian.org/debian bullseye-updates InRelease
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Err:6 tor+https://deb.debian.org/debian-security bullseye-security InRelease
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Err:7 tor+https://deb.debian.org/debian bullseye-backports InRelease
  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
Reading package lists... Done                        
E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye-updates/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch tor+https://deb.debian.org/debian-security/dists/bullseye-security/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch tor+https://deb.debian.org/debian/dists/bullseye-backports/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch tor+https://fasttrack.debian.net/debian/dists/bullseye-fasttrack/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch tor+https://deb.whonix.org/dists/bullseye/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Failed to fetch https://deb.qubes-os.org/r4.1/vm/dists/bullseye/InRelease  Invalid response from proxy: HTTP/1.0 500 Unable to connect  Server: tinyproxy/1.10.0  Content-Type: text/html  Connection: close     [IP: 127.0.0.1 8082]
E: Some index files failed to download. They have been ignored, or old ones used instead.
[WARNING] [systemcheck] Debian Package Update Check Result: Could not check for software updates! (apt-get code: 100)
Please manually check inside this TemplateVM ('whonix-gw-16').
1. Open a terminal. (dom0 -> Start Menu -> Template: whonix-gw-16 -> Terminal)
2. Update. upgrade-nonroot
[INFO] [systemcheck] Warrant Canary Check: systemd unit canary not running.
[INFO] [systemcheck] Please donate!
   See: https://www.whonix.org/wiki/Donate

**Comment by bluesteal** [comment]
sys-whonix Entire output:

[INFO] [systemcheck] sys-whonix-clone-1 | Whonix-Gateway | whonix-gw-16 TemplateBased ProxyVM | Sun 06 Aug 2023 07:00:13 PM UTC
[INFO] [systemcheck] Check sudo Result: OK
[INFO] [systemcheck] Whonix build version: 3:8.1-1
[INFO] [systemcheck] whonix-gateway-packages-dependencies-cli: 23.0-1
[INFO] [systemcheck] derivative_major_release_version /etc/whonix_version: 16
[INFO] [systemcheck] Whonix Support Status of this Major Version: Ok.
[WARNING] [systemcheck] Hardened Malloc: Disabled.
[INFO] [systemcheck] Spectre Meltdown Test: skipping since spectre_meltdown_check=false, ok.
[INFO] [systemcheck] Package Manager Consistency Check Result: Output of command dpkg --audit was empty, ok.
[INFO] [systemcheck] systemd journal check Result:
AppArmor:
########################################

########################################

secomp:
########################################

########################################

warnings:
########################################
Aug 06 16:44:29 host tor[856]: Aug 06 16:44:29.270 [warn] Option 'DisableNetwork' used more than once; all but the last value will be ignored.
Aug 06 16:44:29 host tor[866]: Aug 06 16:44:29.451 [warn] Option 'DisableNetwork' used more than once; all but the last value will be ignored.
Aug 06 17:09:36 host qubes.StartApp+xfce4-terminal-dom0[2421]: (xfce4-terminal:2425): dbind-WARNING **: 17:09:36.605: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Aug 06 17:10:16 host qubes.StartApp+xfce4-terminal-dom0[2421]: (xfce4-terminal:2425): Gtk-WARNING **: 17:10:16.121: Could not load a pixbuf from /org/gtk/libgtk/theme/Adwaita/assets/check-symbolic.svg.
Aug 06 17:14:58 host qubes-gui[850]: WARNING handle_focus: Window 0xe00003 data not initializedgot unknown msg type 131, ignoring
Aug 06 18:44:54 host vanguards[857]: WARNING[Sun Aug 06 18:44:54 2023]: Circ 301 exceeded CIRC_MAX_HSDESC_KILOBYTES: 35121 > 30720.
Aug 06 18:59:52 host qubes.StartApp+xfce4-terminal-dom0[3381]: (xfce4-terminal:3385): dbind-WARNING **: 18:59:52.306: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
########################################

failed:
########################################
Aug 06 16:45:11 host kernel: tsc: Fast TSC calibration failed
Aug 06 16:45:11 host kernel: Failed to register legacy timer interrupt
Aug 06 16:45:13 host swapon[368]: swapon: /dev/xvdc1: swapon failed: Device or resource busy
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:47 host sdwdate[915]: 2023-08-06 16:44:47 - sdwdate - INFO - failed_urls: 0 allowed_failures: 7
Aug 06 16:45:51 host dbus-daemon[1303]: [session uid=1000 pid=1301] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
Aug 06 17:09:36 host qubes.StartApp+xfce4-terminal-dom0[2421]: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
Aug 06 17:10:06 host sudo[2514]:     user : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/usr/bin/systemctl list-units --failed
Aug 06 18:52:12 host sdwdate[915]: 2023-08-06 18:52:12 - sdwdate - INFO - failed_urls: 0 allowed_failures: 7
Aug 06 18:59:52 host qubes.StartApp+xfce4-terminal-dom0[3381]: Failed to connect to session manager: Failed to connect to the session manager: SESSION_MANAGER environment variable not defined
Aug 06 19:00:15 host sudo[4369]: systemcheck : TTY=pts/0 ; PWD=/home/user ; USER=root ; COMMAND=/bin/systemctl --no-pager --no-block --no-legend --failed
########################################

errors:
########################################
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 00, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 01, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 03, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 04, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 05, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 06, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: ACPI Error: No handler or method for GPE 07, disabling event (20220331/evgpe-839)
Aug 06 16:45:11 host kernel: RAS: Correctable Errors collector initialized.
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:28 host xl[654]: libxl: error: libxl_utils.c:815:libxl_cpu_bitmap_alloc: failed to retrieve the maximum number of cpus
Aug 06 16:44:28 host sdwdate-pre[628]: + gcc /usr/libexec/sdwdate/sclockadj.c -o /usr/libexec/sdwdate/sclockadj -ldl -D_GNU_SOURCE -Wdate-time -D_FORTIFY_SOURCE=3 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now
Aug 06 17:09:36 host qubes.StartApp+xfce4-terminal-dom0[2421]: (xfce4-terminal:2425): dbind-WARNING **: 17:09:36.605: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Aug 06 18:59:52 host qubes.StartApp+xfce4-terminal-dom0[3381]: (xfce4-terminal:3385): dbind-WARNING **: 18:59:52.306: AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
########################################

ordering cycle:
########################################

########################################

To see this for yourself...
1. Open a terminal. (dom0 -> Start Menu -> ServiceVM: sys-whonix-clone-1 -> Terminal)
3. Run. sudo apparmor-info
3. Run. sudo journalctl --boot | grep -i syscall=
4. Run. sudo journalctl --boot | grep -i warn
5. Run. sudo journalctl --boot | grep -i fail
6. Run. sudo journalctl --boot | grep -i error
7. Run. sudo journalctl --boot | grep -i "ordering cycle"

If you know what you are doing, feel free to disable this check.
Create a file /etc/systemcheck.d/50_user.conf and add:
systemcheck_skip_functions+=" check_journal "
[INFO] [systemcheck] check network interfaces Result: Ok.
[INFO] [systemcheck] Qubes qubes-db Test Result: Connection to local qubes-db daemon succeeded, ok.
[INFO] [systemcheck] Qubes Settings Test Result: Ok. (GATEWAY_IP: 127.0.0.1)
[INFO] [systemcheck] Qubes Settings Test Result: Ok, qubes_vm_type is ProxyVM.
[INFO] [systemcheck] Check Kernel Messages Test Result: Found nothing remarkable, ok.
[INFO] [systemcheck] Whonix firewall systemd unit check Result: Ok.
[INFO] [systemcheck] Check setup-dist Result: done, ok.
[INFO] [systemcheck] Check Package Manager Running Result: None running, ok.
[INFO] [systemcheck] Tor Check Result: "DisableNetwork 1" not active, ok.
[INFO] [systemcheck] Tor Config Check Result: Tor config ok.
[INFO] [systemcheck] Tor Running Check Result: Pid '866' running.
[INFO] [systemcheck] Control Port Filter Proxy Test Result: OK
[INFO] [systemcheck] check_anondate_do debugging information:

tor_consensus_status       : verified
current_time_in_valid_range: true

current_time_torish        : 2023-08-06 19:00:18
tor_consensus_valid_after  : 2023-08-06 17:00:00
tor_consensus_valid_until  : 2023-08-06 20:00:00
tor_consensus_middle_range : 2023-08-06 18:30:00

tor_cert_lifetime_output   : 
tor_cert_lifetime_valid    : true
tor_cert_valid_after       : 

tor_consensus_user_permission : debian-tor
tor_consensus_group_permission: debian-tor
[INFO] [systemcheck] Tor SocksPort Reachability Test Result: Reachable. (curl exit code: 22 | curl status message: [22] - [HTTP page not retrieved. The requested url was not found or returned another error with the HTTP error code being 400 or above. This return code only appears if -f, --fail is used.])
[INFO] [systemcheck] Tor Connection Result: Ok.
tor_bootstrap_status: NOTICE BOOTSTRAP PROGRESS=100 TAG=done SUMMARY="Done"
Tor Circuit: established
[INFO] [systemcheck] Time Synchronization Result: Ok.
Time synchronization status: success
sdwdate reports: Success.
whonix_firewall status: consecutive run after boot
onion-time-pre-script reports: 
__ ### START: ### /usr/libexec/helper-scripts/onion-time-pre-script
__ Status: Subsequent run after boot.
__ Static Time Sanity Check: Within minimum time 'Fri Feb 18 00:00:00 UTC 2022' and expiration timestamp 'Tue May 17 10:00:00 UTC 2033', ok.
__ Tor reports: NOTICE BOOTSTRAP PROGRESS=100 TAG=done SUMMARY="Done"
__ Tor circuit: established
__ Tor Consensus Time Sanity Check: Clock within consensus parameters consensus/valid-after 2023-08-06 17:00:00 and consensus/valid-until 2023-08-06 20:00:00.
__ Conclusion: Tor already reports circuit established.
__ ### END: ### Exiting with exit_code '0' indicating 'success'.
[INFO] [systemcheck] Connected to Tor.
[INFO] [systemcheck] Whonix Meta Packages Test Result: Meta package qubes-whonix-gateway installed, ok.
[INFO] [systemcheck] Whonix Unwanted Packages Test Result: None found.
[INFO] [systemcheck] Check Initializer Result: /var/lib/initializer-dist/status-files/first_run_initializer.fail does not exist, ok.
[INFO] [systemcheck] Check Virtualizer Result: Supported Virtualizer Qubes detected, continuing.
systemd-detect-virt result: qubes
[INFO] [systemcheck] PVClock Result: /sys/devices/system/clocksource/clocksource0/current_clocksource exist, is xen. This is non-ideal. A known issue. Contributions happily considered. Read more: https://phabricator.whonix.org/T389
[INFO] [systemcheck] Check Timezone Result: /etc/timezone, Etc/UTC matches Etc/UTC, ok.
[INFO] [systemcheck] Check Timezone Result: /usr/share/zoneinfo/Etc/UTC matches /etc/localtime, ok.
[INFO] [systemcheck] IP Forwarding Result: /proc/sys/net/ipv4/ip_forward is 0, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /var/lib/systemcheck/.msgcollector/msgdispatcher-error.log does not exist, ok.
[INFO] [systemcheck] Check Logs Result: /run/systemcheck/.cache/tb/torbrowser_updater_error.log does not exist, ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --fqdn" output is "host.localdomain", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname" output is "host", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --ip-address" output is "127.0.0.1", ok.
[INFO] [systemcheck] Check Hostname Result: "hostname --ip-address" output is "localdomain", ok.
[INFO] [systemcheck] Entropy Available Check Result: ok. /proc/sys/kernel/random/entropy_avail: 256
[INFO] [systemcheck] Check nonfree Result: Ok, no nonfree packages found. For more information, see:
https://www.whonix.org/wiki/Avoid_nonfree_software
[INFO] [systemcheck] Whonix APT Repository: Enabled.
When the Whonix team releases BULLSEYE updates,
they will be AUTOMATICALLY installed (when you run apt-get dist-upgrade)
along with updated packages from the Debian team. Please
read https://www.whonix.org/wiki/Trust to understand the risk.
If you want to change this, use:
    sudo whonix_repository
[INFO] [systemcheck] Qubes Update Proxy Reachability Test: Trying to reach Qubes update proxy... PROXY_SERVER: http://127.0.0.1:8082/
[INFO] [systemcheck] Qubes Update Proxy Reachability Result: Ok, torified update proxy reachable.
[INFO] [systemcheck] Qubes Update Proxy Functionality Test: Trying to use Qubes update proxy... PROXY_SERVER: http://127.0.0.1:8082/
[INFO] [systemcheck] Qubes Update Proxy Functionality Result: Ok, functional.
[INFO] [systemcheck] SocksPort IP Leak Test: Testing Tor's SocksPort (SOCKS_PORT_SYSTEMCHECK: 9110)...
[INFO] [systemcheck] SocksPort IP Leak Test: Connected to Tor. 
[INFO] [systemcheck] system default networking IP Leak Test: system default networking IP Leak Test...
[INFO] [systemcheck] system default networking IP Leak Test Test Result: Ok, Tor's TransPort was not reachable.
This is expected on Whonix-Gateway's default configuration.
(curl exit code: [6] - [Couldn't resolve host. The given remote host was not resolved.])
[INFO] [systemcheck] Stream Isolation Test: Ok, skipped, because TransPort test failed! Can not test stream isolation. This is expected on Whonix-Gateway's default configuration.
[INFO] [systemcheck] Debian Package Update Check: Checking for software updates via apt-get... ( Documentation: https://www.whonix.org/wiki/Update )
Hit:1 tor+https://deb.whonix.org bullseye InRelease                                                                                                                         
Hit:2 tor+https://deb.debian.org/debian bullseye InRelease                                           
Hit:3 tor+https://deb.debian.org/debian bullseye-updates InRelease             
Hit:4 tor+https://deb.debian.org/debian-security bullseye-security InRelease   
Get:5 tor+https://fasttrack.debian.net/debian bullseye-fasttrack InRelease [12.9 kB]
Hit:6 tor+https://deb.debian.org/debian bullseye-backports InRelease           
Hit:7 https://deb.qubes-os.org/r4.1/vm bullseye InRelease
Fetched 12.9 kB in 5s (2,773 B/s)
Reading package lists... Done
[INFO] [systemcheck] sudo apt-get dist-upgrade --simulate output:
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages were automatically installed and are no longer required:
  dctrl-tools dkms libblas3 libgfortran5 libimagequant0 liblapack3 libopengl0
  libpython2-stdlib libpython2.7-minimal libpython2.7-stdlib libxen-4.8
  python2 python2-minimal python2.7 python2.7-minimal python3-numpy
  python3-pil
Use 'sudo apt autoremove' to remove them.
The following packages will be upgraded:
  libjavascriptcoregtk-4.0-18 libpam-systemd libsystemd0 libudev1
  libwebkit2gtk-4.0-37 linux-libc-dev systemd systemd-sysv udev
9 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Inst libsystemd0 [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64]) [systemd:amd64 ]
Conf libsystemd0 (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64]) [systemd:amd64 ]
Inst libpam-systemd [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64]) [systemd:amd64 ]
Inst systemd [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Inst udev [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64]) []
Inst libudev1 [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Conf libudev1 (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Conf systemd (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Inst systemd-sysv [247.3-7+deb11u2] (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Inst libwebkit2gtk-4.0-37 [2.40.3-2~deb11u1] (2.40.5-1~deb11u1 Debian-Security:11/oldstable-security [amd64]) []
Inst libjavascriptcoregtk-4.0-18 [2.40.3-2~deb11u1] (2.40.5-1~deb11u1 Debian-Security:11/oldstable-security [amd64])
Inst linux-libc-dev [5.10.179-1] (5.10.179-3 Debian-Security:11/oldstable-security [amd64])
Conf libpam-systemd (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Conf udev (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Conf systemd-sysv (247.3-7+deb11u4 Debian:11-updates/oldstable-updates [amd64])
Conf libwebkit2gtk-4.0-37 (2.40.5-1~deb11u1 Debian-Security:11/oldstable-security [amd64])
Conf libjavascriptcoregtk-4.0-18 (2.40.5-1~deb11u1 Debian-Security:11/oldstable-security [amd64])
Conf linux-libc-dev (5.10.179-3 Debian-Security:11/oldstable-security [amd64])
[WARNING] [systemcheck] Debian Package Update Check Result: apt-get reports that packages can be updated.
Please update your 'whonix-gw-16' TemplateVM.
1. Open a TemplateVM terminal. (dom0 -> Start Menu -> Template: whonix-gw-16 -> Terminal)
2. Update. upgrade-nonroot
3. Shutdown your TemplateVM. (dom0 -> Qubes VM Manager -> right click 'whonix-gw-16' -> Shutdown VM)
4. Shutdown and restart this TemplateBased ProxyVM. (dom0 -> Qubes VM Manager -> right click 'sys-whonix-clone-1' -> Shutdown VM)
[INFO] [systemcheck] Warrant Canary Check: Debugging information:
Canary Download Time  : Sun Aug  6 01:15:25 UTC 2023
Current Time          : Sun Aug  6 19:00:44 UTC 2023
Canary Signature Time : Sat Aug  5 14:50:54 UTC 2023
Canary Expiration Time: Sat Sep  2 14:50:54 UTC 2023
[INFO] [systemcheck] Warrant Canary Check: Ok.
[INFO] [systemcheck] Please donate!
   See: https://www.whonix.org/wiki/Donate

**Comment by marmarek** [comment]
So, it seems from sys-whonix perspective that everything is fine, but the template see an error from tinyproxy.

You mentioned you have two sys-whonix, can you check which one gets used for UpdatesProxy connections? The easiest way is checking `journalctl` in dom0 and looking for lines with `qubes.UpdatesProxy`, for example this way:
```
journalctl -e |grep qubes.UpdatesProxy
```

**Comment by bluesteal** [comment]
journalctl -e |grep qubes.UpdatesProxy

This command did not return anything in dom0

**Comment by marmarek** [comment]
Even just after calling systemcheck in whonix template? `-e` option displays last 1000 lines, if access to updates proxy was earlier, it wouldn't show up, you can increase number of lines with `-n` option

**Comment by bluesteal** [comment]
this is in dom0 only right?

**Comment by bluesteal** [comment]
Ok this command -> journalctl -n 10000 |grep qubes.UpdatesProxy returns:

dom0 qrexec-policy-daemon[1976]: qrexec: qubes.UpdatesProxy+: whonix-gw-16 -> @default: allowed to sys-whonix


I think this means it is using sys-whonix as the default update proxy but this needs to be changed, how can I do this?

**Comment by marmarek** [comment]
Yes, `qubes.UpdatesProxy` lines are logged in dom0. It should look like this:
```
Jul 08 16:30:32 dom0 qrexec-policy-daemon[2251]: qrexec: qubes.UpdatesProxy+: whonix-gw-16 -> @default: allowed to sys-whonix
```

**Comment by marmarek** [comment]
> needs to be changed, how can I do this?

See https://www.whonix.org/wiki/Qubes/UpdatesProxy

**Comment by bluesteal** [comment]
For some reason, there is no /etc/qubes-rpc/policy/qubes.UpdatesProxy file. Should I just create one?

**Comment by bluesteal** [comment]
Or should I use the Salt formula instead?

**Comment by marmarek** [comment]
Is that qubes 4.1 or 4.2? This sounds like a 4.2, right? If so, you can change that using graphical "Qubes global config" tool - in the "updates" tab, there is "Whonix update proxy" option

**Comment by bluesteal** [comment]
This is R4.1.2; does that tool exist in that version?

**Comment by marmarek** [comment]
No. In that case, you can either create the file manually, or use Salt formula.

**Comment by bluesteal** [comment]
I used the Salt formula and then modified the qubes.UpdatesProxy file to reflect the cloned sys-whonix. This was successful.
Thank you kindly.

**Comment by adrelanos** [comment]
@bluesteal 
> I think this means it is using sys-whonix as the default update proxy but this needs to be changed, how can I do this?

See documentation:
https://www.whonix.org/wiki/Multiple_Whonix-Gateway

(Was just very recently re-checked and updated.)

**Comment by adrelanos** [comment]
@bluesteal
> I used the Salt formula and then modified the qubes.UpdatesProxy file to reflect the cloned sys-whonix. This was successful.

Do you mean the issue has been resolved, you Whonix Template updates are now functional for you?

**Comment by bluesteal** [comment]
there is an issue with reinstalling whonix templates but it is a separate issue.

**Comment by andrewdavidwong** [comment]
So, is there any action to be taken here on the part of the Whonix and/or Qubes devs? Did this turn out not to be a bug?

**Comment by adrelanos** [comment]
I suggest to close this until/if the user replies.
